src/layer/OpenSearchLayer.js

```diff
@@ -0,0 +1,370 @@
+         * @return {Promise} A promise which when resolved returns this layer, or an error when rejected
+         * @example openSearchLayer
+         *                      .discover({url: 'http://example.com/opensearch'})
+         *                      .then(result => console.log(result))
+         *                      .catch(err => console.error(err));
+         */
+        OpenSearchLayer.prototype.discover = function (options) {
+            var self = this;
+            return this._searchService.discover(options)
+                .then(function () {
+                    return self;
+                });
+        };
+
+        /**
+         * Performs a search query.
+         *
+         * @param {Object[]|null} searchParams The list of search parameters, each object must have a name and value
+         * property.
+         * @param {OpenSearchRequest|null} options See {@link OpenSearchRequest} for possible options.
+         * @return {Promise} A promise which when resolved returns a geoJSON collection, or an error when rejected.
+         * @example openSearchService
+         *                      .search([
+         *                          {name: 'count', value: 50}, {name: 'lat', value: 50}, {name: 'lon', value: 20}
+         *                      ])
+         *                      .then(result => console.log(result))
+         *                      .catch(err => console.error(err));
+         */
+        OpenSearchLayer.prototype.search = function (searchParams, options) {
+            var self = this;
+
+            if (!options || options.replaceShapes !== false) {
+                this.removeAllRenderables();
+            }
+
+            return this._searchService.search(searchParams, options)
+                .then(function (geoJSONCollection) {
+                    self.loadFromGeoJSON(geoJSONCollection, self._shapeConfigurationCallback);
+                    self.filterByDate();
+                    return geoJSONCollection;
+                });
+        };
+
+        /**
+         * Loads a GeoJSON object in this layer.
+         *
+         * @param {Object} geoJSONCollection The GeoJSON collection to load.
+         * @param {Function} shapeConfigurationCallback The optional function called prior to creating a shape for
+         * the indicated GeoJSON geometry.
+         * This function can be used to assign attributes to newly created shapes.
+         * The callback function's first argument is the current geometry object.
+         * The second argument to the callback function is the object containing the properties read from the
+         * corresponding GeoJSON properties member, if any.
+         */
+        OpenSearchLayer.prototype.loadFromGeoJSON = function (geoJSONCollection, shapeConfigurationCallback) {
+            var shapeCb = shapeConfigurationCallback || this._shapeConfigurationCallback || null;
+            var polygonGeoJSON = new GeoJSONParser(JSON.stringify(geoJSONCollection));
```

Comment:
We stringify the object here for having the parser convert it back right away. Please prepare a separate pull request adding objects as an option of the GeoJSONParser in addition to strings. Then, leverage this new option here.